api/src/main/java/org/openmrs/module/pihmalawi/patient/impl/YendaNafePatientRegistrationServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+import org.openmrs.PatientIdentifierType;
+import org.openmrs.Person;
+import org.openmrs.PersonAddress;
+import org.openmrs.PersonName;
+import org.openmrs.User;
+import org.openmrs.annotation.Authorized;
+import org.openmrs.api.LocationService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.PersonService;
+import org.openmrs.api.UserService;
+import org.openmrs.module.pihmalawi.models.YendaNafePatientRegistrationModel;
+import org.openmrs.module.pihmalawi.patient.YendaNafePatientRegistrationService;
+import org.openmrs.module.reporting.common.DateUtil;
+import org.openmrs.util.PrivilegeConstants;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.SortedSet;
+import java.util.TreeSet;
+/**
+ *  Class implements patient creation operations for patients captured from medic mobile
+ */
+@Component
+public class YendaNafePatientRegistrationServiceImpl implements YendaNafePatientRegistrationService {
+    @Authorized( { PrivilegeConstants.ADD_PATIENTS })
+    @Transactional
+    @Override
+    public Patient createPatient(YendaNafePatientRegistrationModel yendaNafePatientRegistrationModel, String creatorUUID, PersonService personService, PatientService patientService, UserService userService, LocationService locationService,String patientIdentifierUuid) {
+        User user = userService.getUserByUuid(creatorUUID);
+
+        String fullName = yendaNafePatientRegistrationModel.name;
+        String[] nameSplitted = fullName.split("\\s+",2);
+        String firstName = nameSplitted[0];
+        String lastName = nameSplitted[1];
+        String gender;
+        if(yendaNafePatientRegistrationModel.sex.toLowerCase().trim().equals("male"))
+        {
+            gender = "M";
+        }
+        else if(yendaNafePatientRegistrationModel.sex.toLowerCase().trim().equals("female"))
+        {
+            gender = "F";
+        }
+        else
+        {
+            gender = "U";
+        }
+
+        SortedSet<PersonName> personNames = new TreeSet<PersonName>();
+
+        PersonName personName = new PersonName();
+        personName.setGivenName(firstName);
+        personName.setFamilyName(lastName);
+        personNames.add(personName);
+
+        SortedSet<PersonAddress> personAddresses = new TreeSet<PersonAddress>();
+        PersonAddress personAddress = new PersonAddress();
+        personAddress.setCountyDistrict("Malawi");
+        personAddress.setCountyDistrict("Neno");
+        personAddress.setCityVillage(yendaNafePatientRegistrationModel.village);
+        personAddresses.add(personAddress);
+
+        Person newPerson = new Person();
+        newPerson.setGender(gender);
+        newPerson.setNames(personNames);
+        newPerson.setCreator(user);
+        newPerson.setBirthdate(DateUtil.parseYmd(yendaNafePatientRegistrationModel.date_of_birth));
+        newPerson.setAddresses(personAddresses);
+        newPerson.addName(personName);
```

Comment:
You call "addName" here, but "setNames" above.  This will likely add 2 of the same name to each patient.  I would get rid of the sortedset altogether, and just use the addName method once.

Reply:
Thanks @mseaton for noting that, I will work on it.